akka-stream/src/main/scala/akka/stream/impl/Sinks.scala

```diff
@@ -0,0 +1,195 @@
+/**
+ * Copyright (C) 2014 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.impl
+
+import java.util.concurrent.atomic.AtomicReference
+import akka.actor.{ ActorRef, Props }
+import akka.stream.impl.StreamLayout.{ Mapping, Module }
+import akka.stream.scaladsl.OperationAttributes._
+import akka.stream.scaladsl.{ Sink, OperationAttributes, Source }
+import akka.stream.stage._
+import akka.stream.{ Inlet, Outlet, InPort, OutPort }
+import org.reactivestreams.{ Processor, Publisher, Subscriber, Subscription }
+import scala.annotation.unchecked.uncheckedVariance
+import scala.concurrent.{ Future, Promise }
+import scala.util.{ Failure, Success, Try }
+import akka.stream.SinkShape
+
+trait SinkModule[-In, Mat] extends StreamLayout.Module {
+
+  /**
+   * This method is only used for Sinks that return true from [[#isActive]], which then must
+   * implement it.
+   */
+  def create(materializer: ActorBasedFlowMaterializer, flowName: String): (Subscriber[In] @uncheckedVariance, Mat)
```

Comment:
should we have a source comment at each place we use uncheckedVariance explaining the reason for its usage